libcxx/include/flat_set

```diff
@@ -0,0 +1,913 @@
+  static_assert(__is_cpp17_random_access_iterator<typename _KeyContainer::const_iterator>::value);
+
+public:
+  using key_type = _Key;
+  using value_type = _Key;
+  using key_compare = _Compare;
+  using value_compare = _Compare;
+  using reference = _Key&;
+  using const_reference = const _Key&;
+  using size_type = typename _KeyContainer::size_type;
+  using difference_type = typename _KeyContainer::difference_type;
+  using iterator = typename _KeyContainer::const_iterator;
+  using const_iterator = typename _KeyContainer::const_iterator;
+  using reverse_iterator = std::reverse_iterator<iterator>;
+  using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+  using container_type = _KeyContainer;
+
+  // [flat.set.cons]
+
+  flat_set()
+    noexcept(is_nothrow_default_constructible_v<container_type> && is_nothrow_default_constructible_v<key_compare>)
+    = default;
+
+  explicit flat_set(container_type __cont)
+    : __c_(std::move(__cont)) {
+    auto __begin = __c_.begin();
+    auto __end = __c_.end();
+    std::stable_sort(__begin, __end, __comp_ref());
+    __c_.erase(std::__flatset_unique(__begin, __end, __comp_ref()), __end);
+  }
+
+  template <class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(const container_type& __cont, const _Alloc& __a)
+    : __c_(std::__make_obj_using_allocator<container_type>(__a, __cont)) {
+    auto __begin = __c_.begin();
+    auto __end = __c_.end();
+    std::stable_sort(__begin, __end, __comp_ref());
+    __c_.erase(std::__flatset_unique(__begin, __end, __comp_ref()), __end);
+  }
+
+  template <class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(container_type&& __cont, const _Alloc& __a)
+    : __c_(std::__make_obj_using_allocator<container_type>(__a, std::move(__cont))) {
+    auto __begin = __c_.begin();
+    auto __end = __c_.end();
+    std::stable_sort(__begin, __end, __comp_ref());
+    __c_.erase(std::__flatset_unique(__begin, __end, __comp_ref()), __end);
+  }
+
+  flat_set(container_type __cont, const key_compare& __comp)
+    : __compare_(__comp), __c_(std::move(__cont)) {
+    auto __begin = __c_.begin();
+    auto __end = __c_.end();
+    std::stable_sort(__begin, __end, __comp_ref());
+    __c_.erase(std::__flatset_unique(__begin, __end, __comp_ref()), __end);
+  }
+
+  template <class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(const container_type& __cont, const key_compare& __comp, const _Alloc& __a)
+    : __compare_(__comp), __c_(std::__make_obj_using_allocator<container_type>(__a, __cont)) {
+    auto __begin = __c_.begin();
+    auto __end = __c_.end();
+    std::stable_sort(__begin, __end, __comp_ref());
+    __c_.erase(std::__flatset_unique(__begin, __end, __comp_ref()), __end);
+  }
+
+  template <class _Alloc>
+    requires uses_allocator_v<container_type, _Alloc>
+  flat_set(container_type&& __cont, const key_compare& __comp, const _Alloc& __a)
```

Comment:
Same, pending LWG issue.

Reply:
https://wg21.link/lwg3803 , fixed in a separate commit with "LWG3803" in the subject line.